### PR TITLE
Networking group notes from 2023.08.24

### PR DESCRIPTION
**Pull request by alexandruradovici**
### Pull Request Overview

This pull request adds the notes from the august 24th networking group meeting.

### Testing Strategy

N/A

### TODO or Help Wanted

The drawings used in the meeting still have to be added.

### Documentation Updated

- [ ] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [ ] Ran `make prepush`.

### CODE REVIEW

**Review by alevy — DISMISSED** [review]

**Comment by alevy** [comment]
Do we need two approvers for meeting notes? @tock/core-wg

**Comment by lschuermann** [comment]
@alevy Precedent says "no", but if it only has one or two I usually wait a couple of days for others to have a chance to look at them.

File: doc/wg/network/notes/network-notes-2023-08-24.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    4. Leon: Tock-ethernet integration strategy
+- **References:**
+    - [NRF52840 802.15.4 PR](https://github.com/tock/tock/pull/3615)
+    - [STM32F429 Ethernet PR](https://github.com/tock/tock/pull/3523)
+
+## Updates
+- Branden: There's a label for Network WG related PRs on the Tock github repo
+- Tyler: Working on 15.4 PR. Need some closure on behavior for transmit
+- Branden: We had a discussion about handling on/off states for the 15.4 radio. Other things in Tock like GPIO just "do the right thing" and set themselves in the state they need to be. The 15.4 radio HIL expects callbacks when changing power states though, which likely makes sense for external radio chips and radios that take some real-world time to start up.
+- [TODO Fill in notes from recording]
```
Comment (lschuermann) [inline doc/wg/network/notes/network-notes-2023-08-24.md:24]: TODO

**Review by lschuermann — CHANGES_REQUESTED** [review]
One TODO still left and missing the draw.io images. Let me take care of this as I promised.

**Comment by ppannuto** [comment]
Thanks for the great notes and apologies I haven't been able to make a call yet.

One thing I wanted to flag for consideration in the "does UDP, TCP support live in capsule or userspace" is the implication of core kernel operations / services possibly relying on the userspace networking stacks (e.g., a kernel updater service is necessarily a very sensitive service).

This is really part of a much bigger and more fundamental permission / system management model that Tock has largely managed to punt on to date—what is responsible for platform management: a supervisor process (à la `init` in Linux), some kind of management capsule, board/platform authors (hello process reset policy questions). While the bigger discussion is more of a core team question, there is a fundamental question here of whether we want to _require_ that core kernel services trust userspace for some fairly fundamental operations (i.e. any kind of real-world networking). Without thinking too deeply, kernel-based UDP and userspace TCP might be one reasonable balance point, but this is a fairly deep question long-term (maybe even thesis-worthy, something between a macro and micro kernel, ... etc :) ).

File: doc/wg/network/notes/network-notes-2023-08-24.md (hunk 1 above)
Comment (brghena) [reply doc/wg/network/notes/network-notes-2023-08-24.md:24]: Reminder to handle these TODOs @lschuermann

Comment (lschuermann) [reply doc/wg/network/notes/network-notes-2023-08-24.md:24]: Yup sorry! I had some more time critical things come up, but I'll try to have this done today.

Comment (brghena) [reply doc/wg/network/notes/network-notes-2023-08-24.md:24]: No rush! Just wanted to remind you it exists.

**Comment by alevy** [comment]
> kernel-based UDP and userspace TCP might be one reasonable balance point

Yes. Doing TCP well is quite a complex state machine and doing it generally is even more complex. It's also probably seldom necessary

> possibly relying on the userspace networking stacks (e.g., a kernel updater service is necessarily a very sensitive service).

Yes, but it being sensitive does not necessarily mean it ought to be in the kernel. In fact, a pretty reasonable design might be to have the communication part take place in userspace, and once downloading a new image is done, have some bootloader or kernel component validate and setup switching to the new image. In other words, decouple the relatively dangerous task of speaking some complex protocol over the network from actually validating and updating the kernel.

**Comment by lschuermann** [comment]
I'm unable to push to the source branch of this PR, but I've pushed my changes here:
https://github.com/tock/tock/blob/25e3c869201d727c98c18e3c055e726b8fd15f05/doc/wg/network/notes/network-notes-2023-08-24.md
Maybe you can pull those commits (on branch `network-notes-2023-08-24`) in, @alexandruradovici.

**Comment by alexandruradovici** [comment]
Done

**Comment by brghena** [comment]
@alexandruradovici You also need to pull this commit from Leon for text updates: https://github.com/tock/tock/commit/9eafa639ea473e0150e17357b1a15584cb815fc3

File: doc/wg/network/notes/network-notes-2023-08-24.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+- Leon: One option is to move the UDP/TCP boxes into the Thread component here, so the socket is accessed through the Thread interface. So when a socket is specific for a given underlying interface, it might make sense to make the UDP component instantiated within the Thread capsule. Then Ethernet could conceptually instantiate its own UDP component
+- Amit: Right now, in the 6LoWPAN stack there is a UDP component. If I'm not mistaken, it's not explicitly a system call driver (rather a library for constructing UDP packets). What's the imagined purposes of having UDP and (in particular) TCP in the kernel at all?
+- Tyler: You're proposing UDP and TCP would be moved outside of the kernel?
+- Amit: TCP seems hard to implement in the kernel. In userspace, there is dynamic memory allocation and libraries, etc.
+
+  Is the purpose of having them in the kernel purely for abstraction purposes? Or perhaps also for ACL reasons?
+- Alex: It seems there is another reason - some hardware peripherals expose UDP and CTP sockets directly. So having support in the kernel would let us work with those or with other boards.
```
Comment (lschuermann) [inline doc/wg/network/notes/network-notes-2023-08-24.md:81]:
```suggestion
- Alex: It seems there is another reason - some hardware peripherals expose UDP and TCP sockets directly. So having support in the kernel would let us work with those or with other boards.
```

Don't really know how that happened...

**Review by lschuermann — APPROVED** [review]

**Review by brghena — APPROVED** [review]

**Pull request merged**